assignment/impl/src/java/org/sakaiproject/assignment/impl/sort/AssignmentSubmissionComparator.java

```diff
@@ -62,11 +66,45 @@ public AssignmentSubmissionComparator(AssignmentService assignmentService, SiteS
 
     @Override
     public int compare(AssignmentSubmission a1, AssignmentSubmission a2) {
-        int result;
+        // Compare by UserSortNameComparator if appropriate
+        Optional<User> u1 = getIndividualSubmitterAsUser(a1);
+        Optional<User> u2 = getIndividualSubmitterAsUser(a2);
+        if (u1.isPresent() && u2.isPresent()) {
+            return userSortNameComparator.compare(u1.get(), u2.get());
+        }
+
         String name1 = getSubmitterSortname(a1);
         String name2 = getSubmitterSortname(a2);
-        result = compareString(name1, name2);
-        return result;
+        return compareString(name1, name2);
+    }
+
+    /**
+     * Gets the submitter of a submission for a non-group assignment.
+     * @return Optional<User> of the submission's submitter.
+     * Returns Optional.empty() if:
+     *     the submission is associated with a group assignment
+     *     the submission does not have exactly 1 associated submitter
+     *     the submitter cannot be retrieved via the UserDirectoryService
+     */
+    private Optional<User> getIndividualSubmitterAsUser(AssignmentSubmission submission) {
+        if (submission.getAssignment().getIsGroup()) {
+            return Optional.empty();
+        }
+
+        Set<AssignmentSubmissionSubmitter> submitters = submission.getSubmitters();
+        if (submitters.size() != 1) {
+            // On a non-group assignment, there should be precisely one submitter.
+            // When instructors submit on behalf of a user, it does not create a new submitter entry; rather it adds a submission property.
+            log.warn("Submission for a non-group assignment has multiple submitters. SubmissionID: " + submission.getId());
```

Comment:
uses older log style please use the newer style {}

Reply:
Sounds good, I'll update this in the morning

Reply:
Log style is updated